tap_core/src/tap_manager/manager.rs

```diff
@@ -181,11 +190,12 @@ impl<
     fn collect_receipts(
         &mut self,
         timestamp_buffer_ns: u64,
+        min_timestamp_ns: u64,
     ) -> Result<(Vec<SignedReceipt>, Vec<SignedReceipt>), Error> {
-        let cutoff_timestamp = crate::get_current_timestamp_u64_ns()? - timestamp_buffer_ns;
+        let max_timestamp_ns = crate::get_current_timestamp_u64_ns()? - timestamp_buffer_ns;
         let received_receipts = self
             .receipt_storage_adapter
-            .retrieve_receipts_upto_timestamp(cutoff_timestamp)
+            .retrieve_receipts_in_timestamp_range(min_timestamp_ns..max_timestamp_ns)
```

Comment:
It seems `max_timestamp_ns` can be less than `min_timestamp_ns` depending on `timestamp_buffer_ns`. I think that means the range passed in will be empty, so `self
            .receipt_storage_adapter
            .retrieve_receipts_in_timestamp_range(min_timestamp_ns..max_timestamp_ns)` should be an empty Vec? 
            
Is that behavior okay? Should there be some kind of warning or error that max timestamp is less than min?

Reply:
Thats a good point. We can probably throw an error in that case.